geps/gep-1619.md

```diff
@@ -5,6 +5,21 @@
 
 (See definitions in [GEP Status][/contributing/gep#status].)
 
+## Graduation Criteria for Implementable Status
+
+This GEP was accidentally merged as Provisional before the required approval
+from 2 maintainers had been received. Before this graduates to implementable,
+we need to get at least one of @robscott or @youngnick to also approve this GEP.
+
+There are some open questions that will need to be answered before this can
```

Comment:
Yeah good question. The things I'm trying to highlight are potentially big API changes that we need to figure out before moving to implementable. IE if we're increasing the potential scope of the resource, that probably means a new name, maybe an enum field for a type of persistence, etc. Really just an overall plan for how this policy resource could grow to support related use cases in the future.

On the other hand, I think that many of the "open questions" in the section below this could likely be addressed at a future point. Here's a quick run through of my thoughts on each of them:

- What happens when session persistence is broken because the backend is not up or healthy? If that's an error case, how should that be handled? Should the API dictate the http error code? Or should the API dictate fall back behavior?

Would really need to look at what implementations are capable of here. Probably worth answering prior to release, but the answer may be that we can't dictate anything if behavior ends up being fairly impl-specific here.

- What happens when session persistence causes traffic splitting scenarios to overload a backend?

This seems like something we likely can't prevent, but we may need to document the risks of persistence. Similar to above, this may end up being a documentation-only thing.

- Should we add status somewhere when a user gets in to a "risky" configuration with session persistence?

I'm not sure how we'd define these risky states, but this feels like it could be handled with a follow up change so should not block.

- Should there be an API configuration field that specifies how already established sessions are handled?

Not sure I get this one? Is this trying to ask what would happen to existing sessions as session persistence is being enabled or disabled? I _think_ this is referring to the potential addition of a new field, so I think this is also probably good criteria for a non-blocking follow up, but may be misunderstanding.

Reply:
Thanks for your thought process, it makes sense now that you used the ones with the highest level of impact to the GEP.

> Not sure I get this one? Is this trying to ask what would happen to existing sessions as session persistence is being enabled or disabled?

Correct, more so on the enabled side-of-things. An API field to control how to handle an existing session with the same name, whether it should forcibly "override" or "rewrite", or it should be passive and not enable persistence if there is a collision.

> I think this is referring to the potential addition of a new field, so I think this is also probably good criteria for a non-blocking follow up, but may be misunderstanding.

Correct, that's fair.